sdk/trace/src/main/scala/org/typelevel/otel4s/sdk/trace/SdkSpanBackend.scala

```diff
@@ -0,0 +1,277 @@
+      resource = immutableState.resource
+    )
+
+  def hasEnded: F[Boolean] =
+    mutableState.get.map(_.hasEnded)
+
+  def latencyNanos: F[Long] =
+    for {
+      state <- mutableState.get
+      endEpochNanos <-
+        if (state.hasEnded) Monad[F].pure(state.endEpochNanos)
+        else Clock[F].realTime.map(_.toNanos)
+    } yield endEpochNanos - immutableState.startEpochNanos
+
+  def getAttribute[A](key: AttributeKey[A]): F[Option[A]] =
+    for {
+      state <- mutableState.get
+    } yield state.attributes.get(key).map(_.value)
+}
+
+object SdkSpanBackend {
+
+  private final case class ImmutableState(
+      name: String,
+      context: SpanContext,
+      scopeInfo: InstrumentationScopeInfo,
+      kind: SpanKind,
+      parentContext: Option[SpanContext],
+      resource: Resource,
+      links: List[LinkData],
+      totalRecordedLinks: Int,
+      startEpochNanos: Long
+  )
+
+  private final case class MutableState(
+      status: StatusData,
+      attributes: Attributes,
+      events: List[EventData],
+      totalRecordedEvents: Int,
+      endEpochNanos: Long,
+      hasEnded: Boolean
+  )
+
+  // todo: replace Concurrent with Async
+  def start[F[_]: Concurrent: Clock](
+      context: SpanContext,
+      name: String,
+      scopeInfo: InstrumentationScopeInfo,
+      resource: Resource,
+      kind: SpanKind,
+      //  parentSpan: Span[F],
+      parentContext: Option[SpanContext],
+      spanLimits: SpanLimits,
+      spanProcessor: SpanProcessor[F],
+      attributes: Attributes,
+      links: List[LinkData],
+      totalRecordedLinks: Int,
+      userStartEpochNanos: Long
+  ): F[SdkSpanBackend[F]] = {
+
+    /*val (clock: Clock[F], createdAnchoredClock: Boolean) =
+      if (parentSpan.isInstanceOf[Span[F]]) {
+        (parentSpan.clock, false)
+      } else {
+        (AnchoredClock.create(tracerClock), true)
+      }*/
+
+    val createdAnchoredClock = false
+    val clock = Clock[F]
+
+    val computeNow =
+      if (userStartEpochNanos != 0) {
+        Monad[F].pure(userStartEpochNanos)
+      } else if (createdAnchoredClock) {
+        clock.realTime.map(_.toNanos) // clock.startTime
+      } else {
+        clock.realTime.map(_.toNanos)
+      }
+
+    def immutableState(startEpochNanos: Long) =
+      ImmutableState(
+        name = name,
+        context = context,
+        scopeInfo = scopeInfo,
+        kind = kind,
+        parentContext = parentContext,
+        resource = resource,
+        links = links,
+        totalRecordedLinks = totalRecordedLinks,
+        startEpochNanos = startEpochNanos
+      )
+
+    val mutableState = MutableState(
+      status = StatusData.Unset,
+      attributes = attributes,
+      events = Nil,
+      totalRecordedEvents = 0,
+      endEpochNanos = 0,
+      hasEnded = false
+    )
+
+    for {
+      start <- computeNow
+      ms <- Concurrent[F].ref(
+        mutableState
+      ) // todo: use AtomicCell[F].of(mutableState)
```

Comment:
A weird thing. If I use `AtomicCell[F].of(mutableState)`, the module compiles correctly, but the `sbt doc` step fails with:
```
/home/runner/work/otel4s/otel4s/sdk/trace/src/main/scala/org/typelevel/otel4s/sdk/trace/SdkSpanBackend.scala:261:29: could not find implicit value for parameter F: cats.effect.kernel.Async[F]
[error]       ms <- AtomicCell[F].of(mutableState)
```

I guess the old cats-effect version is selected in the `unidocs` project and that's why the `doc` step fails.

Reply:
@armanbilge have you seen something similar in other projects?

Reply:
Yes, this looks like https://github.com/sbt/sbt-unidoc/issues/77. A workaround is to override the classpath with one from another project e.g. https://github.com/armanbilge/calico/commit/4230c1120cee6de476d460c8ad25fddc2ce86aba